### ISSUE TITLE
Support Relu activation function with CUDA cuDNN gpu accleeration  instead of just tanh

### ISSUE THREAD

**Issue by Cpt-Falcon** [tensorflow/tensorflow#46065]
**System information**
- TensorFlow version (you are using):
   2.4
- Are you willing to contribute it (Yes/No):
Yes

**Describe the feature and the current behavior/state.**

The docs say that cuDNN requires tanh for some reason although its not clear why this limitation exists:

https://www.tensorflow.org/api_docs/python/tf/keras/layers/GRU#used-in-the-notebooks_1

Tensorflow should support more activation functions like Relu when using cuDNN.  On the Nvidia cuDNN side, there is no limitation with using relu

https://developer.nvidia.com/blog/optimizing-recurrent-neural-networks-cudnn-5/

Right now I am being forced to use tanh instead of relu simply because of the fact that the performance of cuDNN with tanh is like 5 orders of magnitude better for training in terms of speed compared to the generic kernel. 

In summary, please remove the validation and requirements check for cuDNN that prevents you from using relu. Thank you.

**Will this change the current api? How?**
It should not affect the API at all

**Who will benefit with this feature?**
Huge performance benefits if able to use cuDNN instead of the generic tensorflow kernel, so anyone with an nvidia GPU would benefit.
**Any Other info.**

**Comment by WindQAQ** [comment]
The cuDNN docs say that it supports four types of rnn cell, and the **relu activation rnn cell** is one of it. However, the relu activation GRU cell is not included in cuDNN. You can see [cudnnRNNMode](https://docs.nvidia.com/deeplearning/cudnn/api/index.html#cudnnRNNMode_t) for the more detailed fomula of each cell type.

**Comment by tilakrayal** [comment]
@Cpt-Falco,
Have you got the chance to have a look at this official document https://docs.nvidia.com/deeplearning/cudnn/latest/api/cudnn-graph-library.html#cudnnRNNMode_t.


Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The Tensorflow team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the Tensorflow space.
